sourced/ml/cmd_entries/repos2bow.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from uuid import uuid4
+
+from scipy import sparse
+
+from sourced.ml.extractors import __extractors__
+from sourced.ml.models import BOW, OrderedDocumentFrequencies, DocumentFrequencies
+from sourced.ml.transformers import Engine, UastExtractor, UastDeserializer, Documents2DocFreq, \
+    Documents2BOW, HeadFiles
+from sourced.ml.utils import create_engine
+
+
+def repos2bow_entry(args):
+    log = logging.getLogger("repos2bow")
+    engine = create_engine("repos2bow-%s" % uuid4(), **args.__dict__)
+    document_column_name = "repository_id"
+    extractors = [__extractors__[s](
+        args.min_docfreq, **__extractors__[s].get_kwargs_fromcmdline(args))
+        for s in args.feature]
+
+    uast = Engine(engine, explain=args.explain) \
+        .link(HeadFiles()) \
+        .link(UastExtractor(languages=args.languages)) \
+        .link(UastDeserializer())
+    df = uast.link(Documents2DocFreq(extractors, document_column_name))
```

Comment:
We can add Waren's quantizer here too

Reply:
you mean like an option?

Reply:
Obligatory, some extractors require an additional "quantization" step